ethereum/c-kzg-4844
# Reorganize/Clean-Up Node Bindings

This branch is submitted as a suggested fix for #149.  Most of this stuff is related to TS/JS and npm publishing so I have asked @dapplion to take a peek.

## Inclusions
- create `src`, `lib` and `test` folders
- move respective files into their folders
- modify `bundle` command to fully create bundle for structure verification prior to publish
- modify `publish` command to only publish package post visual inspection (and potentially testing)
- remove `.npmignore` (no longer needed because of bundle command change)
- tightened the typescript configuration to be much more strict 
- remove babel, rollup and their configs (not needed. doesn't need to be bundled nor poly-filled for server side JS)
- add industry standard `bindings` package to `require` the `.node` file instead of trying to manually do it
- remove post-build copy step in bindings.gyp (using package above removes the need)
- added my name as a contributor to `package.json` (should we also add the names of people that contributed to the base library as well?)

## Testing Instructions
```bash
git remote add matthewkeil git@github.com:matthewkeil/c-kzg-4844.git
git fetch matthewkeil reorganize-node-bindings
cd bindings/node.js
yarn install --ignore-scripts
make
cd dist
yarn install  # is simulating what happens when a consumer installs the bundle as a dependency
```

## Notes
- Tests are failing.  Fix is in PR #188.  I merged in fix locally and tested before writing this PR though.  Will come back and triple check everything is still working as expected after the prior PR's are merged.
